### ISSUE TITLE
Switch off the anchor links feature by default

### ISSUE THREAD

**Issue by kaedroho** [wagtail/wagtail#6963]
I know that this is an existing feature. However, the original PR wasn't discussed in an RFC and there wasn't any kind of design review, so I think we should discuss this.

I think the "anchor links" feature could be quite confusing for editors who work on sites that don't actually support anchor links. I find this to be quite a rare requirement but it is always switched on by default.

I propose that we disable it by default to reduce clutter/potential confusion for the majority of sites that do not use it. It should be easy for a developer to find a setting to switch it back on as they have to build support into their templates in order for this feature to work anyway.

Also, there are a couple of bugs with it:
 - When you add an anchor link in the editor, clicking the link takes the user back to the page editor with the `#` set. It should take the user to the page on the site.
 - It's missing from the [editor's guide](https://docs.wagtail.io/en/stable/editor_manual/new_pages/inserting_links.html)

**Comment by benjaoming** [comment]
I agree that the current default is bad and raises more questions than what it contributes. Is there a way to make the feature useful by design?

* Ensuring that anchor (header id) support is easy to switch on, such that all core widgets render header ids?
* Adding support for anchor links on internal pages?
* Detect broken anchor links when validating page contexts? Or a management script to check links?

I can see that this quickly snowballs. But the feature is very useful :)
